[Issue] panic: runtime error: invalid memory address or nil pointer dereference | ACA Deployment

**Output from `azd version`**
azd version 1.10.2

**Describe the bug**
This has two aca deployments and after it finishes the second deployment, it shows a failure. However, the app is deployed.

**To Reproduce**
Run azd up on this repo or run the gh action: https://github.com/Azure-Samples/contoso-creative-writer/

**Expected behavior**
The deployment completes without errors.

**Additional context**
gh action failure here: https://github.com/Azure-Samples/contoso-creative-writer/actions/runs/11374746532/job/31644084161

Run azd deploy --no-prompt
Deploying services (azd deploy)
Deploying service api
Deploying service api (Packing remote build context)
Deploying service api (Uploading remote build context)
Deploying service api (Updating container app revision)
Deploying service api (Fetching endpoints for container app service)
  (✓) Done: Deploying service api
  - Endpoint: https://agent-api.canadaeast.azurecontainerapps.io/
Deploying service web
Deploying service web (Packing remote build context)
Deploying service web (Uploading remote build context)
Deploying service web (Updating container app revision)
Deploying service web (Fetching endpoints for container app service)
  (✓) Done: Deploying service web
  - Endpoint: https://agent-web.canadaeast.azurecontainerapps.io/
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x88 pc=0x560353df1d1f]
goroutine 1 [running]:
github.com/azure/azure-dev/cli/azd/pkg/ext.(*HooksManager).filterConfigs(0xc0001[25](https://github.com/Azure-Samples/contoso-creative-writer/actions/runs/11374746532/job/31644084161#step:8:26)e20, 0xc0005d6ea8?, 0xc0005d6e08)
	/mnt/vss/_work/1/s/cli/azd/pkg/ext/hooks_manager.go:92 +0x1bf
github.com/azure/azure-dev/cli/azd/pkg/ext.(*HooksManager).GetByParams(0xc000125e20, 0xc0002850e0, {0x56035417ee13, 0x4}, {0xc0005d7098, 0x1, 0x560353650fa5?})
	/mnt/vss/_work/1/s/cli/azd/pkg/ext/hooks_manager.go:66 +0x1fa
github.com/azure/azure-dev/cli/azd/pkg/ext.(*HooksRunner).RunHooks(0xc0001b4de0, {0x560354aa5b48, 0xc000284750}, {0x56035417ee13?, 0x1?}, 0x0, {0xc0005d7098, 0x1, 0x1})
	/mnt/vss/_work/1/s/cli/azd/pkg/ext/hooks_runner.go:89 +0x7b
github.com/azure/azure-dev/cli/azd/pkg/ext.(*HooksRunner).Invoke(0xc0001b4de0, {0x560354aa5b48, 0xc000284750}, {0xc0005d7098, 0x1, 0x1}, 0xc0000bb0b8)
	/mnt/vss/_work/1/s/cli/azd/pkg/ext/hooks_runner.go:74 +0xf5
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*HooksMiddleware).registerCommandHooks(0xc00011b7c0, {0x560354aa5b48, 0xc000284750}, 0xc0002854a0, 0xc00002f700, 0xc0003d4dc0)
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/hooks.go:105 +0x2d5
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*HooksMiddleware).Run(0xc00011b7c0, {0x560354aa5b48, 0xc000284750}, 0xc0003d4dc0)
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/hooks.go:66 +0x1be
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x560354aa5b48, 0xc000284750})
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/middleware.go:109 +0x2b8
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*TelemetryMiddleware).Run(0xc000124dc0, {0x560354aa5b48, 0xc0000256e0}, 0xc0003d4dc0)
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/telemetry.go:78 +0x8ee
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x560354aa5b48, 0xc0000256e0})
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/middleware.go:109 +0x2b8
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*UxMiddleware).Run(0xc0000127b0, {0x560354aa5b48, 0xc0000256e0}, 0xc0003d4dc0)
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/ux.go:32 +0xa9
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x560354aa5b48, 0xc0000256e0})
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/middleware.go:109 +0x2b8
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*DebugMiddleware).Run(0xc000012780, {0x560354aa5b48, 0xc0000256e0}, 0xc0003d4dc0)
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/debug.go:48 +0x382
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction.func1({0x560354aa5b48, 0xc0000256e0})
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/middleware.go:109 +0x2b8
github.com/azure/azure-dev/cli/azd/cmd/middleware.(*MiddlewareRunner).RunAction(0xc00046d520, {0x560354aa5b48, 0xc0000256e0}, 0xc0001b4ba0, {0xc0001[30](https://github.com/Azure-Samples/contoso-creative-writer/actions/runs/11374746532/job/31644084161#step:8:31)0c0, 0x11})
	/mnt/vss/_work/1/s/cli/azd/cmd/middleware/middleware.go:130 +0x137
github.com/azure/azure-dev/cli/azd/cmd.(*CobraBuilder).configureActionResolver.func1(0xc0004cd8c8, {0xc000124b80, 0x0, 0x1})
	/mnt/vss/_work/1/s/cli/azd/cmd/cobra_builder.go:130 +0x386
github.com/spf13/cobra.(*Command).execute(0xc0004cd8c8, {0xc000124b70, 0x1, 0x1})
	/home/cloudtest/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x68a
github.com/spf13/cobra.(*Command).ExecuteC(0xc000490dc8)
	/home/cloudtest/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x[38](https://github.com/Azure-Samples/contoso-creative-writer/actions/runs/11374746532/job/31644084161#step:8:39)d
github.com/spf13/cobra.(*Command).Execute(...)
	/home/cloudtest/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/spf13/cobra.(*Command).ExecuteContext(...)
	/home/cloudtest/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:895
main.main()
	/mnt/vss/_work/1/s/cli/azd/main.go:65 +0x24b
Error: Process completed with exit code 2.

--- a/cli/azd/pkg/ext/hooks_manager.go
+++ b/cli/azd/pkg/ext/hooks_manager.go
@@ -2,6 +2,7 @@ package ext
 
 import (
 	"fmt"
+	"log"
 	"os"
 	"runtime"
 	"strings"
@@ -81,6 +82,10 @@ func (h *HooksManager) filterConfigs(
 		}
 
 		for _, hook := range hooks {
+			if hook == nil {
+				log.Printf("hook configuration for '%s' is missing", scriptName)
+				continue
+			}
 
 			if predicate != nil && !predicate(scriptName, hook) {
 				continue

--- a/cli/azd/pkg/ext/hooks_manager_test.go
+++ b/cli/azd/pkg/ext/hooks_manager_test.go
@@ -62,6 +62,20 @@ func Test_GetAllHookConfigs(t *testing.T) {
 		require.Nil(t, validHooks)
 		require.Error(t, err)
 	})
+
+	t.Run("With Missing Configuration", func(t *testing.T) {
+		// All hooksMap are invalid because they are missing a script type
+		hooksMap := map[string][]*HookConfig{
+			"preprovision": nil,
+		}
+
+		hooksManager := NewHooksManager(tempDir)
+		validHooks, err := hooksManager.GetAll(hooksMap)
+
+		require.NoError(t, err)
+		require.NotNil(t, validHooks)
+		require.Len(t, validHooks, 0)
+	})
 }
 
 func Test_GetByParams(t *testing.T) {

